bin/eslint_d.js

```diff
@@ -54,7 +60,7 @@ const command = process.argv[2];
       (await import('../lib/status.js')).status(resolver, config);
       return;
     default:
-      if (config && config.hash !== hash) {
+      if (config && (config.hash !== hash || !(await isAlive(config)))) {
```

Comment:
Using `isAlive` in the start and status commands is a very good improvement. But calling it here adds an extra roundtrip to the server on every lint call. I ran some quick tests and it slows it down quite a bit. Given that this is the hot code execution path, I'd like to not do that here.

There is a good alternative though: The `forwardToDaemon` function already handles connection errors and removes the config. Instead of doing that in the function itself, `forwardToDaemon` could return a flag or throw an exception, so that this file here could handle connection issues, restart the service and try sending the request again. This would eliminate the need to call `isAlive` with the same behaviour.

Am I missing something? What do you think?

Reply:
Agree, I think using optimistic approach is good idea at least because this check will return "true" most of the time